uni/lib/controller/local_storage/app_shared_preferences.dart

```diff
@@ -117,6 +118,17 @@ class AppSharedPreferences {
     return prefs.setInt(themeMode, (themeIndex + 1) % 3);
   }
 
+  static Future<void> setLocale(String appLocale) async {
+    final prefs = await SharedPreferences.getInstance();
+    await prefs.setString(locale, appLocale);
+  }
+
+  static Future<Locale> getLocale() async {
+    final prefs = await SharedPreferences.getInstance();
+    final appLocale = prefs.getString(locale) ?? 'en';
```

Comment:
Debatable question
Should the default be "pt_PT"?

Reply:
The default should be portuguese if the phone is in Portuguese, otherwise english imo

Reply:
How do we know the phone default?

Reply:
But I agree

Reply:
https://stackoverflow.com/questions/50923906/how-to-get-timezone-language-and-county-id-in-flutter-by-the-location-of-device